rfc-000-frontend-fewer-apps.md

```diff
@@ -0,0 +1,256 @@
+Each application will be analysed before being consolidated to ensure redundant code is not being moved over.
+
+### How the proposal will be carried out
+This is obviously a very large piece of work that will take some time to complete. We have identified the following steps that will be needed.
+
+* Prepare `frontend` to be the host
+    * audit `frontend` and make changes to the frontend directory structure to match “new layout”
+    * remove redundant code from frontend (non-blocking)
+    * remove publishing_api tasks from frontend (non-blocking)
+
+* Write a roadmap for app consolidation
+    * will be handled separately
+    * might be simply the order in which applications will be merged into the host
+    * or could be some apps merged together before merging into the host
+    * or some routes merging ahead of others
+* Consider and work on expected issues (see below)
+* Work to merge remaining apps
+
+### Steps as part of a merge
+We anticipate that any merge will likely involve some or all of the following steps.
+
+* Identify any issues with existing code
+    * size these issues
+    * either fix issues or record them as tech debt - we need to avoid trying to fix everything and concentrate on progressing consolidation
+* Replicate the rendering from the candidate to the host
+    * preserve the git history of the candidate app - add a permalink to the files being consolidated in the commit in the host app.
+* Change content api
+    * change the rendering app for the document type in the publishing app.
+    * republish content
+    * republish any special routes (things that don’t have a publishing app, like the homepage) - any special routes should be moved into special route publisher (we have an outstanding tech debt card relating to this)
+* Optimise the frontend code
+    * remove duplication
+* Remove the routes from the candidate
+* [Retire the candidate] once all routes have been removed
+
+### Expected issues
+Merging these four apps is not without difficulties and we acknowledge that a single app will present problems that our multiple apps do not.
+
+* A Rails update for one larger app might be more complex than spread across individual apps.
+* The scaling of a single app might become more costly.
+* It won’t be possible to scale targeted parts of the stack (i.e. one of the existing apps) for expected increases in traffic.
+* It’s only possible to test one branch of code in a test environment at once. If there’s only one repo for these four apps, only one change can be tested at a time.
+* It will be harder to find the pull request history for the retired repositories.
+
+In addition, we will need to address the following issues:
+
+* The test suite will take longer due to more tests.
+    * could potentially parallelise in CI
+    * will need to consider approach for developer machines
+* Need to reconfigure Kubernetes deployment. We should be able to drop the total number of pods.
```

Comment:
In RFC-174 Sean commented that the increase in cost is negligible, so we probably don't need to bother with this.

Reply:
Have removed scaling cost references.